### PR TITLE
Annotate Microsoft.CSharp

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to #2339
Tagging @stephentoub @jaredpar 

Annotations were determined by a combination of checking doc comments and examining usage.

### CODE REVIEW

File: src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.csproj [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
+    <Nullable>enable</Nullable>
+    <NoWarn>$(NoWarn);nullable</NoWarn>
```
Comment (stephentoub) [inline src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.csproj:5]: Do nullable warnings need to be suppressed in the ref?  I'd have expected this to only be needed in the src.

File: src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.cs [hunk 2]
```diff
@@ -59,14 +59,14 @@ public partial class RuntimeBinderException : System.Exception
...
+        public RuntimeBinderException(string? message, System.Exception? innerException) { }
     }
     public partial class RuntimeBinderInternalCompilerException : System.Exception
     {
         public RuntimeBinderInternalCompilerException() { }
         protected RuntimeBinderInternalCompilerException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
-        public RuntimeBinderInternalCompilerException(string message) { }
-        public RuntimeBinderInternalCompilerException(string message, System.Exception innerException) { }
+        public RuntimeBinderInternalCompilerException(string? message) { }
+        public RuntimeBinderInternalCompilerException(string? message, System.Exception? innerException) { }
```
Comment (stephentoub) [inline src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.cs:70]: For such a large library, so little surface area :)

**Review by stephentoub — APPROVED** [review]
Just spot checked, but other than my one question, LGTM. Thanks!

File: src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.csproj (hunk 1 above)
Comment (RikkiGibson) [reply src/libraries/Microsoft.CSharp/ref/Microsoft.CSharp.csproj]: There are no warnings to suppress, so I removed this property.

**Review by stephentoub — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]
I have a lot of `?` about this PR

**Pull request merged**